### ISSUE TITLE
Add calibration validation metrics and residual visualization

### ISSUE THREAD

**Issue by Nuno-Monteiro** [GoCleverOrg/poc-homography#138]
## Summary
Add hold-out validation, comprehensive error metrics, and residual visualization to the GCP-based calibration system. This prevents "it works sometimes" scenarios by providing quantitative validation of calibration quality through train/test splits, percentile error reporting, spatial residual plots, and systematic error pattern detection.

## Context
Current calibration in `gcp_calibrator.py` optimizes camera parameters but lacks formal validation infrastructure. The system computes RMS error on training data but provides no hold-out validation to detect overfitting, no visualization of residual spatial patterns to identify systematic errors (wrong distortion, axis misalignment), and no quantitative thresholds for calibration acceptance. This gap makes it difficult to distinguish between good and poor calibrations until runtime failures occur.

This issue is part of the geo-registration system overhaul (Phase 5) to establish rigorous calibration quality assurance.

### In Scope
- Split GCPs into train (70-80%) and test (20-30%) sets for hold-out validation
- Compute validation metrics on test set: RMS error, 90th percentile error, max error
- Generate residual visualizations: spatial distribution overlaid on image, magnitude histogram
- Visualize reprojection error vectors as arrows (observed to predicted pixels)
- Detect systematic error patterns: directional bias (all residuals point same direction), radial growth (errors increase toward image edges)
- Define pass/fail thresholds for calibration acceptance based on RMS and percentile errors
- Generate per-frame and aggregate validation reports
- Extend `CalibrationResult` to include validation metrics

### Out of Scope
- Modifying the core optimization algorithm in `GCPCalibrator.calibrate()`
- Adding new calibration parameters beyond existing pan/tilt/roll/position
- Real-time validation during optimization iterations
- Automatic recalibration when validation fails
- Integration with web UI tools (`unified_gcp_tool.py`, `capture_gcps_web.py`)
- Lens distortion calibration

## Definition of Done
- [ ] `GCPCalibrator` accepts `validation_split` parameter (default 0.2) to reserve test GCPs
- [ ] Calibration optimizes on train set only, evaluates on test set
- [ ] `CalibrationResult` includes `validation_metrics` dict with RMS, 90th percentile, and max errors for both train and test sets
- [ ] Function `detect_systematic_errors()` returns warnings for directional bias and radial growth patterns
- [ ] Function `generate_residual_plot()` creates matplotlib figure with residual arrows overlaid on image coordinates
- [ ] Function `generate_residual_histogram()` creates matplotlib figure showing error magnitude distribution
- [ ] Function `validate_calibration()` returns pass/fail based on configurable RMS threshold (default ≤5 pixels)
- [ ] Per-GCP validation report includes individual errors and inlier/outlier classification
- [ ] Unit tests verify train/test split, metric computation, and systematic error detection
- [ ] Property-based tests verify validation metrics are consistent with optimization results

### Hold-Out Validation

Implement train/test splitting for GCP-based calibration to detect overfitting.

**Context**: Current calibration evaluates error on the same GCPs used for optimization, which can hide overfitting to training data. Hold-out validation provides unbiased error estimates (standard practice in machine learning validation).

**Requirements**:
- Accept `validation_split` parameter in `GCPCalibrator.__init__()` (float in [0.0, 0.5], default 0.2)
- Randomly split GCPs into train (optimization) and test (validation) sets
- Use fixed random seed for reproducibility
- Require minimum GCP count: at least 6 GCPs for train set, at least 3 for test set
- If insufficient GCPs, skip splitting and warn (validate on training set)
- Store train/test indices in `CalibrationResult` for transparency

**Reference**: Existing `_compute_residuals()` method in `gcp_calibrator.py` (line 350) computes per-GCP errors. Extend to compute separately for train and test subsets.

### Validation Metrics

Compute comprehensive error statistics beyond RMS.

**Context**: RMS error alone masks outliers. 90th percentile and max error expose worst-case performance critical for safety-critical applications.

**Requirements**:
- Compute for both train and test sets: RMS error, 90th percentile error, max error
- Store in `CalibrationResult.validation_metrics` dictionary:
  ```python
  {
      'train': {'rms': float, 'p90': float, 'max': float, 'count': int},
      'test': {'rms': float, 'p90': float, 'max': float, 'count': int}
  }
  ```
- Use `np.percentile(errors, 90)` for 90th percentile
- Ensure per-GCP errors are Euclidean distances (already computed in line 580)

**Reference**: Existing `per_gcp_errors` in `CalibrationResult` (line 105). Extend to separate train/test errors.

### Systematic Error Detection

Identify calibration failure modes through residual pattern analysis.

**Context**: Certain calibration errors produce characteristic residual patterns:
- Directional bias (all residuals point same direction) → wrong camera height, tilt, or pan axis
- Radial growth (errors increase from center to edges) → wrong lens distortion model or principal point

**Requirements**:
- Implement `detect_systematic_errors(residuals_2d: np.ndarray, image_coords: np.ndarray) -> List[str]`
- Directional bias detection: compute mean residual vector, check if magnitude > threshold (e.g., 2 pixels)
- Radial growth detection: compute correlation between error magnitude and distance from image center, flag if correlation > 0.5
- Return list of warning strings describing detected patterns
- Call from `calibrate()` method after optimization completes

**Example**:
```python
warnings = detect_systematic_errors(residuals_2d, image_coords)
# Returns: ["Directional bias detected: mean residual (2.3, -1.8) suggests incorrect tilt axis"]
```

### Residual Visualization

Generate matplotlib plots showing spatial distribution and magnitude of reprojection errors.

**Context**: Visual inspection of residuals reveals calibration quality and failure modes not captured by scalar metrics. Arrow plots show directionality, histograms show distribution shape.

**Requirements**:
- Implement `generate_residual_plot(calibration_result: CalibrationResult, image_width: int, image_height: int) -> matplotlib.figure.Figure`
  - Scatter plot with image dimensions as axes
  - Draw arrows from observed to predicted pixel coordinates (or predicted to observed, whichever is clearer)
  - Color-code by error magnitude (colorbar)
  - Annotate with GCP indices
  - Show train/test split with different markers
- Implement `generate_residual_histogram(calibration_result: CalibrationResult) -> matplotlib.figure.Figure`
  - Histogram of error magnitudes
  - Separate train/test distributions
  - Mark RMS, 90th percentile, max error as vertical lines
  - Log scale if errors span multiple orders of magnitude

**Reference**: Existing matplotlib usage in `tests/test_gcp_round_trip_validation.py` for plot patterns. Use `matplotlib.pyplot.figure()` for figure creation, `plt.quiver()` for arrow plots.

### Pass/Fail Thresholds

Define quantitative acceptance criteria for calibration quality.

**Context**: Manual judgment of "good enough" calibration is unreliable. Quantitative thresholds enable automated validation and clear acceptance criteria.

**Requirements**:
- Implement `validate_calibration(calibration_result: CalibrationResult, max_rms: float = 5.0, max_p90: float = 10.0) -> Dict[str, Any]`
- Default thresholds: RMS ≤ 5 pixels (tight), 90th percentile ≤ 10 pixels (moderate outliers allowed)
- Evaluate thresholds on TEST set (not train set)
- Return dictionary:
  ```python
  {
      'passed': bool,
      'test_rms': float,
      'test_p90': float,
      'threshold_rms': float,
      'threshold_p90': float,
      'failures': List[str]  # e.g., ["RMS error 6.2px exceeds threshold 5.0px"]
  }
  ```
- Make thresholds configurable parameters

**Reference**: Existing inlier threshold logic in `gcp_calibrator.py` (line 584) uses `loss_scale * 2.0`. Validation thresholds are independent of robust loss scale.

### Validation Reports

Generate human-readable reports summarizing calibration quality.

**Context**: Developers and operators need clear summaries to assess calibration quality without inspecting code or logs.

**Requirements**:
- Per-GCP report: table with columns [GCP index, train/test, observed (u,v), predicted (u,v), error (px), inlier/outlier]
- Aggregate report: summary statistics (train/test metrics, systematic warnings, pass/fail status)
- Format as Markdown for easy integration with documentation
- Implement `generate_validation_report(calibration_result: CalibrationResult) -> str`

**Example Report**:
```
## Calibration Validation Report

### Aggregate Metrics
- Train Set: RMS=2.1px, P90=3.8px, Max=5.2px (N=24)
- Test Set: RMS=2.3px, P90=4.1px, Max=6.0px (N=6)
- Status: PASS (RMS ≤ 5px, P90 ≤ 10px)

### Systematic Error Analysis
- No directional bias detected
- No radial growth detected

### Per-GCP Errors
| Index | Set   | Observed (u,v) | Predicted (u,v) | Error (px) | Status  |
|-------|-------|----------------|-----------------|------------|---------|
| 0     | train | (960, 540)     | (958.2, 541.1)  | 2.2        | inlier  |
...
```

### Testing Strategy

Ensure validation framework is mathematically correct and robust.

**Context**: Validation bugs undermine calibration confidence. Comprehensive tests prevent regression.

**Requirements**:
- Unit tests for train/test splitting (verify counts, no overlap, reproducibility)
- Unit tests for metric computation (known residuals → expected RMS/P90/max)
- Unit tests for systematic error detection (synthetic directional/radial patterns → correct warnings)
- Property-based tests (Hypothesis): validation metrics consistent with optimization results
- Integration test: end-to-end calibration with validation on synthetic GCPs with known ground truth

**Reference**: Existing property-based testing framework in `tests/test_gcp_calibrator_properties.py`. Follow same structure for validation properties.
